CHANGELOG.md

```diff
@@ -1,5 +1,12 @@
 # Changelog
 
+# Version 0.0.0 (YYYY-MM-DD) - In Progress
+
+### Added
+### Changed
+* Removed the `_update_queue_mode` and `get_queue_mode` functions from `Project`
```

Comment:
We generally try avoiding taking away functionality right away because this can lead to user workflows breaking, which in turn makes them go to support and become unhappy.

From what I understand though this is not going to be allowed very soon from the API side. Could we reach out to Alex Cota on a proper deprecation plan for this and time it with the release of a new version to match removing this? 

* We need to notify MLSE/CSMs, to let users know that they will need to tell their customers that this will no longer be supported at date X
* Add deprecation warnings to the functions first, which will have the deprecation date X, and merge that as a PR first

Reply:
The update queue mode functionality currently has undefined behavior when datarows have been attached to a project - this change is critical to preserve customer data integrity. Customers can still specify a queue mode for a project during creation time using the queue_mode attribute, so a migration path is readily available. Product has asked for the removal of this functionality immediately and without any feature flag rollout. That being said, I'll work with Pranoy and Alex to develop a communication plan to minimize surprise.

Reply:
I wonder if it's better to keep the `update queue_mode` functionality in code, but just throw an exception instead, so they are informed right when their script fails, instead of some vague "function missing error". What do you think @richardsun0713 ?